ptf/sairoute.py

```diff
@@ -0,0 +1,1118 @@
+            vr_id=self.default_vrf, destination=sai_ipprefix('10.10.10.1/32'))
+        sai_thrift_create_route_entry(self.client, route_entry1,
+                                      next_hop_id=nhop1)
+
+        route_entry2 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('10.10.10.2/32'))
+        sai_thrift_create_route_entry(self.client, route_entry2,
+                                      next_hop_id=nhop1)
+
+        pkt1 = simple_tcp_packet(
+            eth_dst=ROUTER_MAC,
+            eth_src='00:22:22:22:22:22',
+            ip_dst='10.10.10.1',
+            ip_src='192.168.0.1',
+            ip_id=105,
+            ip_ttl=64)
+        pkt2 = simple_tcp_packet(
+            eth_dst=ROUTER_MAC,
+            eth_src='00:22:22:22:22:22',
+            ip_dst='10.10.10.2',
+            ip_src='192.168.0.1',
+            ip_id=105,
+            ip_ttl=64)
+        exp_pkt1 = simple_tcp_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            ip_dst='10.10.10.1',
+            ip_src='192.168.0.1',
+            ip_id=105,
+            ip_ttl=63)
+        exp_pkt2 = simple_tcp_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            ip_dst='10.10.10.2',
+            ip_src='192.168.0.1',
+            ip_id=105,
+            ip_ttl=63)
+        try:
+            print("Sending packet on port %d, forward" % self.dev_port11)
+            send_packet(self, self.dev_port11, pkt1)
+            verify_packet(self, exp_pkt1, self.dev_port10)
+
+            print("Sending packet on port %d, forward" % self.dev_port11)
+            send_packet(self, self.dev_port11, pkt2)
+            verify_packet(self, exp_pkt2, self.dev_port10)
+
+        finally:
+            sai_thrift_remove_route_entry(self.client, route_entry1)
+            sai_thrift_remove_route_entry(self.client, route_entry2)
+            sai_thrift_remove_neighbor_entry(self.client, neighbor_entry)
+            sai_thrift_remove_next_hop(self.client, nhop1)
+
+    def dropRouteTest(self):
+        '''
+        Verify drop route.
+        '''
+        print("dropRouteTest")
+        dmac = '00:11:22:33:44:55'
+
+        nhop = sai_thrift_create_next_hop(
+            self.client, ip=sai_ipaddress('10.10.10.2'),
+            router_interface_id=self.port10_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+        neighbor_entry = sai_thrift_neighbor_entry_t(
+            rif_id=self.port10_rif, ip_address=sai_ipaddress('10.10.10.2'))
+        sai_thrift_create_neighbor_entry(self.client, neighbor_entry,
+                                         dst_mac_address=dmac)
+
+        route_entry = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf, destination=sai_ipprefix('10.10.10.1/32'))
+        sai_thrift_create_route_entry(
+            self.client, route_entry, next_hop_id=nhop,
+            packet_action=SAI_PACKET_ACTION_TRAP)
```

Comment:
This is probably something we are not supporting or some hack. The test says dropRouteTest but the packet action here is TRAP. Please check.

Reply:
Yes, it was a workaroud because we don't support DROP action in route_entry, I'll fix it.